### PR TITLE
Maximised widgets always force a call to be shown in PIP mode

### PR DESCRIPTION
**Pull request by toger5**
Fixes: https://github.com/vector-im/element-web/issues/19637

---
Here's what your changelog entry will look like:

## ✨ Features
 * Maximised widgets always force a call to be shown in PIP mode ([\#7163](https://github.com/matrix-org/matrix-react-sdk/pull/7163)). Fixes vector-im/element-web#19637. Contributed by @toger5.

Preview: https://619ccdb6e34a10fc35aaec1f--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by gaelledel** [comment]
Screenshot 2021-11-22 at 09 45 03
@toger5 @jryans This is what happens when I tried the build

**Comment by toger5** [comment]
Thanks for trying. There are also failing tests. I definitiv need to investigate this.

**Comment by toger5** [comment]
@gaelledel there was an issue if no room is currently selected which I didn't run into because i never deleted my local storage.
It should be possible to test now. Thanks again for informing me.

File: src/CallHandler.tsx [hunk 1]
```diff
@@ -380,6 +380,16 @@ export default class CallHandler extends EventEmitter {
         return callsNotInThatRoom;
     }
 
+    getAllActiveCallsForPip(roomId) {
```
Comment (turt2live) [inline src/CallHandler.tsx:383]: explicit member access please (public/private)

Comment (turt2live) [inline src/CallHandler.tsx:383]: types on `roomId` and return type.

File: src/components/views/voip/CallPreview.tsx [hunk 2]
```diff
@@ -59,7 +60,9 @@ interface IState {
 // (which should be a single element) of other calls.
 // The primary will be the one not on hold, or an arbitrary one
 // if they're all on hold)
-function getPrimarySecondaryCalls(calls: MatrixCall[]): [MatrixCall, MatrixCall[]] {
+function getPrimarySecondaryCallsForPip(roomId): [MatrixCall, MatrixCall[]] {
```
Comment (turt2live) [inline src/components/views/voip/CallPreview.tsx:63]: types on `roomId`

**Review by turt2live — APPROVED** [review]
otherwise lgtm

**Review by turt2live — APPROVED** [review]

**Comment by Palid** [comment]
@toger5 seems ready to merge, do the honors! 😄

**Pull request merged**